examples/docs/generator/generate.php

```diff
@@ -0,0 +1,199 @@
+        }
+    }
+
+    // Sanitize Filename to Classname
+    $filename  = preg_replace("/[^[:alpha:]]/u", '', ucwords($filename, '-'));
+    $fullpath  = sprintf('./../src/Examples/Docs/%s/%s.php', $subDirs, $filename);
+    $namespace = (empty($parts)) ? '' : '\\' . str_replace('/', '\\', $subDirs);
+
+    // Generate Stub Class
+    if(file_exists($fullpath) === false) {
+        $search  = ['%__NAMESPACE__%', '%__CLASSNAME__%', '%__TIMESTAMP__%', '%__ASCIIDOC__%'];
+        $replace = [$namespace, $filename, date('Y-m-d H:i:s'), $key];
+        $skel    = file_get_contents('./skeletons/class.skel');
+        $class   = str_replace($search, $replace, $skel);
+
+        file_put_contents($fullpath, $class);
+        $counters['new_classes']++;
+    }
+
+    // Bind Examples to Class
+    $class = file_get_contents($fullpath);
+    $fingerprint = md5($class);
+    foreach($examples as $example) {
+        $methodName = sprintf('testExampleL%d_%s', $example['line_no'], $example['hash']);
+
+        // Add new Method
+        if(strpos($class, $methodName) === false) {
+            $curlExample = '';
+            $curlString  = '';
+
+            foreach($example['content'] as $i => $v) {
+                if($i != 0) {
+                    $curlExample .= '        ';
+                    $curlString  .= '              . ';
+                }
+                $curlExample .= sprintf("// %s", $v);
+                $curlString  .= sprintf("'%s'", $v);
+                if($i < count($example['content'])-1) {
+                    $curlExample .= "\n";
+                    $curlString  .= "\n";
+                }
+            }
+
+            $search  = ['%__TAG__%', '%__LINE__%', '%__TIMESTAMP__%', '%__METHOD_NAME__%', '%__CURL_EXAMPLE__%', '%__CURL_STRING__%'];
+            $replace = [$example['hash'], $example['line_no'], date('Y-m-d H:i:s'), $methodName, $curlExample, $curlString];
+            $skel    = file_get_contents('./skeletons/method.skel');
+            $method  = str_replace($search, $replace, $skel);
+            $class   = str_replace('// %__METHOD__%', $method, $class);
+
+            $counters['new_methods']++;
+        }
+    }
+    // Write changes to disk ?
+    if($fingerprint != md5($class)) {
+        file_put_contents($fullpath, $class);
+    }
+
+    // Generate Ascii Doc include files
+    $folders  = (empty($parts)) ? '' : implode('/', $parts) . '/';
+    $filePath = sprintf('%ssrc/Examples/Docs/%s%s.php', str_repeat('../', count($parts)+2), $folders, $filename);
+    $asciidir = sprintf('../asciidoc/%s%s/', $folders, $filename);
+
+    // Build Sub directory Structure
+    if(file_exists($asciidir) === false) {
+        mkdir($asciidir, 0775, true);
+    }
+
+    // Generate Asciifdoc Include Files
+    foreach($examples as $example) {
+        if(file_exists(sprintf('%s/%s.asciidoc', $asciidir, $example['hash'])) === false && file_exists(sprintf('%s/_%s.asciidoc', $asciidir, $example['hash'])) === false) {
```

Comment:
So you make a `_` file which will resolve the include and just remove the `_` when you are ready. Clever.

Reply:
I'm starting to prefer the `.adoc` suffix over `.asciidoc`. Both are standard to be honest and you can totally do whichever.

Reply:
your choice, will rename the files to `.adoc`